### ISSUE TITLE
Divide by zero error when running pytorch_tcorex

### ISSUE THREAD

**Issue by turambar** [Harhro94/Time-Varying-Linear-Corex#3]
Happens on lines [832](https://github.com/Harhro94/Time-Varying-Linear-Corex/blob/master/pytorch_tcorex.py#L832), [839](https://github.com/Harhro94/Time-Varying-Linear-Corex/blob/master/pytorch_tcorex.py#L839), [846](https://github.com/Harhro94/Time-Varying-Linear-Corex/blob/master/pytorch_tcorex.py#L846).

The code on each line is: `w = 1.0 / np.power(self.gamma, np.abs(i - t))`.

When using the default value for gamma (int 2) and for a large exponent, we overflow. A quick fix would be to force gamma to be a float.

**Comment by turambar** [comment]
@Harhro94 Looking through the code on lines 817-823:

```
        window_len = [0] * self.nt
        for i in range(self.nt):
            k = 0
            while k <= self.nt and sum(n_samples[i - k:i + k + 1]) < self.max_sample_count:
                k += 1
            window_len[i] = k
        self.window_len = window_len
```

Is the logic here sound? In my experiments, I call `corex.fit(arr)` with `arr.shape = (639, 12, 47)`. I set `max_sample_cnt=639` when creating `corex`. After the lines above run, the contents of 


I have a time series of length `650`, which I divide into `639` overlapping windows of length `12` (so the input to `fit(...)` has shape `(639, 12, 47)`). I set `max_sample_cnt=639`. After the lines above, the contents of `window_len` are `[346, 346, 346, ..., 27, 27, 27, ..., 28, 29, 30, ..., 53]`. That looks suspicious...
